### ISSUE TITLE
关于带入自己数据集后，train loss大于valid loss大于test loss的疑问

### ISSUE THREAD

**Issue by sunxiaoyao-git** [thuml/Autoformer#155]
您好！
在阅读调试完您的代码后，我带入了自己的数据集，发现loss曲线，训练误差很大，验证误差和测试误差反而小很多。这可能是什么原因造成的呢？会和原始数据的分布有关吗，并且我只用了单变量预测。

**Comment by wuhaixu2016** [comment]
您好，误差与数据本身的尺度有关，因为时间序列的特性会随着时间变化，所以需要具体问题具体分析。

**Comment by sunxiaoyao-git** [comment]
谢谢您！

**Comment by benhsampson** [comment]
This is only part of the reason, I changed the dataset such that train/test/val subset sample randomly from the dataset. That would eliminate any difference between the test/val sets since the model wasn't tuned to fit either. But I still found that the train loss was higher than the test and validation loss at each epoch. This is unusual since the model should fit the training data better than the validation (val) and test data. But I think this is why:
The reason is twofold: (1) The val/test loss is calculated at the end of each epoch, whereas the train loss is calculated *throughout*; so the val/test loss is calculated the latest (~best) model whereas the train loss is calculated by averaging the loss over ~tens of thousands of batches with the model weights changing continuously throughout. (2) Dropout. Dropout randomly drops units (neurons) from the model *during training only*, which is helpful at preventing overfitting, but means the loss will increase as a result.
If you want to confirm this for yourself, set the dropout to 0, and check that the train loss for epoch `n+1` is lower than the val/test loss for epoch `n`. You should find this is true for all epochs `n>=1`.
